src/main/java/macrobase/analysis/summary/Summary.java

```diff
@@ -0,0 +1,7 @@
+package macrobase.analysis.summary;
+
+/**
+ * Represents a summarization result, currently an itemset with a score and weight.
+ */
+public abstract class Summary {
```

Comment:
Yeah, not sure, also, not sure we need a full List<Summary> below. What if we just base it on `List<ItemsetResult>`?